src/adaXT/criteria/criteria.pyx

```diff
@@ -541,9 +542,107 @@ cdef class Linear_regression(Criteria):
             int i, length
 
         length = indices.shape[0]
-        theta0, theta1 = self.theta(indices)
+        theta0, theta1 = self._theta(indices)
         cur_sum = 0.0
         for i in range(length):
             step_calc = self.y[indices[i]] - theta0 - theta1 * self.x[indices[i], 0]
-            cur_sum += step_calc*step_calc
+            cur_sum += step_calc ** 2
```

Comment:
Generally it is less efficient to do something ** 2 instead of doing something*something. The same convention can be seen in scikit learns: https://github.com/scikit-learn/scikit-learn/blob/main/sklearn/tree/_criterion.pyx.

This answer goes more in depth: https://stackoverflow.com/a/72153092.

Reply:
Good point, thanks. Adjusted now.